src/container.ts

```diff
@@ -255,15 +256,28 @@ export class Container {
             return new FixedAssigner('00000000-0000-0000-0000-000000000000');
         }
 
+        const cidKey = 'croct.cid';
+        const cache = this.getLocalStorage();
         const logger = this.getLogger('CidAssigner');
+        let endpoint = this.configuration.cidAssignerEndpointUrl;
+
+        if (this.configuration.refreshCid) {
+            const cachedCid = cache.getItem(cidKey);
+            const baseEndpoint = new URL(endpoint);
+
+            if (cachedCid !== null) {
+                baseEndpoint.searchParams.set('cid', cachedCid);
+                endpoint = baseEndpoint.toString();
+            }
+        }
```

Comment:
If the `refresh` option is used on a client that doesn't have a cached CID, the endpoint constructed here will not contain any CID.
In that case, whenever a CID is requested, the cache will return the last one and request a new value from the inner assigned. The inner assigned will then send a request to this endpoint, which will not receive any value and emit a different CID every time.

This will only work on a remote assigner with working cookies; otherwise, the CID of a single client will change for every call until the second initialization of the SDK.

Reply:
When there's no cached CID, the `CachedCidAssigner` does not refresh the CID:
https://github.com/croct-tech/sdk-js/pull/325/files#diff-801c6cb643fce84879829d2fb850723d6f2ee8b7b9e4456b6a223bf76445c31aR31-R47